cirq-core/cirq/vis/heatmap.py

```diff
@@ -299,6 +300,25 @@ def plot(
             ax = cast(plt.Axes, ax)
         original_config = copy.deepcopy(self._config)
         self.update_config(**kwargs)
+
+        selected_qubits = kwargs.get("selected_qubits")
+        if selected_qubits is not None:
+            edgecolors = tuple(
+                'red' if qubit[0] in selected_qubits else 'grey' \
```

Comment:
not sure why these lines only check the first qubit, should not they check both qubits of the edges? ... also `qubit` is not a good var name .. maybe either `q_pair` or edge

Reply:
This code is inside `Heatmap` instead of `TwoQubitInteractionHeatmap`. `qubit` is essentially a one-tuple entry, so we need to index as `qubit[0]` to get the actual qubit. The value of `sorted(self._value_map.items())` in the two-qubit case is the below:

```
[((cirq.GridQubit(0, 0),), 0.0), ((cirq.GridQubit(0, 1),), 0.0), ((cirq.GridQubit(0, 2),), 0.0), ((cirq.GridQubit(1, 0),), 0.0), ((cirq.GridQubit(3, 2),), 0.0), ((cirq.GridQubit(3, 3),), 0.0)]
```
So the presence of elements such as `(cirq.GridQubit(0, 0),)` makes it imperative to index them as `[0]`.


In this code inside `TwoQubitInteractionHeatmap`:

```py
Heatmap({q: 0.0 for q in qubits}).plot(
            ax=ax,
            collection_options=collection_options,
            plot_colorbar=False,
            annotation_format=None,
            highlighted_qubits=highlighted_qubits,
        )
```

the pair of qubits are "flattened," as seen by `{q: 0.0 for q in qubits}`.

So `qubit[0]` seems a fair choice. I have tried creating a two-qubit example in which the qubit to highlight is the second entry in the tuple. It's able to highlight it.

Reply:
the class member has maps a tuple `self._value_map: Mapping[QubitTuple, SupportsFloat]` so this is the invariant that functions need to respect ... if we decide to implement a new heatmap that doesn't flatten the input and instead assumes that plot does the right thing then your logic breaks

Reply:
> if we decide to implement a new heatmap that doesn't flatten the input and instead assumes that plot does the right thing then your logic breaks

Yes, true. If it's decided later that the input shouldn't be flattened, then the implementation can be changed to loop over a set of tuples instead. I'm unsure if it will be possible or if we even need to rewrite the logic so it remains invariant to new implementations of heatmaps that don't flatten the input.

Reply:
we wouldn't need to reimplement anything if instead of checking `qubit[0]` we do `if any(q in selected_qubits for q in qubits)` (and rename `qubit` to `qubits`)

Reply:
I see, thanks!